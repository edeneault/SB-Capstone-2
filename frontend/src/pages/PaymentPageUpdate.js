import React, { useState } from "react";
import { Form, Button, Col } from "react-bootstrap";
import { useDispatch, useSelector } from "react-redux";
import FormContainer from "../components/FormContainer";
import CheckoutSteps from "../components/CheckoutSteps";
import { savePaymentMethod, updatePaymentMethod } from "../actions/cartActions";

const PaymentScreenUpdate = ({ history }) => {
  const cart = useSelector((state) => state.cart);
  const { shippingAddress } = cart;

  const orderId = useSelector((state) => state.orderDetails.order._id);

  if (!shippingAddress) {
    history.push("/shipping");
  }

  const [paymentMethod, setPaymentMethod] = useState("PayPal");

  const dispatch = useDispatch();

  const submitHandler = (e) => {
    e.preventDefault();
    dispatch(savePaymentMethod(paymentMethod));
    dispatch(updatePaymentMethod(paymentMethod, orderId));
    history.push(`/orders/${orderId}`);
  };

  return (
    <FormContainer>
      <CheckoutSteps step1 step2 step3 />
      <h1>Payment Method</h1>
      <Form onSubmit={submitHandler}>
        <Form.Group>
          <Form.Label as='legend'>Select Method</Form.Label>

          <Col>
            <Form.Check
              type='radio'
              label='Paypal or Credit Card'
              id='Paypal'
              name='paymentMethod'
              value='PayPal'
              checked
              onChange={(e) => setPaymentMethod(e.target.value)}
            ></Form.Check>

            <Form.Check
              type='radio'
              label='Stripe'
              id='Stripe'
              name='paymentMethod'
              value='Stripe'
              onChange={(e) => setPaymentMethod(e.target.value)}
            ></Form.Check>
          </Col>
        </Form.Group>
        <Button type='submit' variant='primary' className='my-2'>
          Continue
        </Button>
      </Form>
    </FormContainer>
  );
};

export default PaymentScreenUpdate;
