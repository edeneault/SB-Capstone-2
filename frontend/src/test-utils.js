import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

// create a customRender that wraps the UI in a memory Router

const MemoryRouterWithInitialRoutes = ({ children, initialRoutes }) => {
  return <MemoryRouter initialEntries={initialRoutes}>{children}</MemoryRouter>;
};

const customRender = (ui, options) => {
  const initialRoutes =
    options && options.initialRoutes ? options.initialRoutes : ["/"];

  return render(ui, {
    wrapper: (args) =>
      MemoryRouterWithInitialRoutes({
        ...args,
        initialRoutes,
      }),
    ...options,
  });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
