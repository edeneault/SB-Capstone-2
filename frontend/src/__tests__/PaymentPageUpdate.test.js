import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { render } from "../test-utils";
import { initialState } from "../store-mock";
import PaymentPageUpdate from "../pages/PaymentPageUpdate.js";
import { createMemoryHistory } from "history";

const mockStore = configureMockStore([thunk]);

describe("test", () => {
  const history = createMemoryHistory();

  test("renders PaymentPageUpdate without crashing", () => {
    const store = mockStore(initialState);

    const { getByText } = render(
      <Provider store={store}>
        <PaymentPageUpdate history={history} />
      </Provider>,
      { initialRoutes: [`/paymentmethod`] },
    );
    // screen.debug();
    expect(getByText(/Payment Method/i)).toBeInTheDocument();
  });

  test("matches snapshot", function () {
    const store = mockStore(initialState);

    const { asFragment } = render(
      <Provider store={store}>
        <PaymentPageUpdate history={history} />
      </Provider>,
      { initialRoutes: [`/paymentmethod`] },
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
