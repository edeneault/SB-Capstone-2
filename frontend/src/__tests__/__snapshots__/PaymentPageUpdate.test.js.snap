// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="px-5 pt-3 fade-in container-fluid"
  >
    <div
      class="container"
    >
      <div
        class="justify-content-md-center row"
      >
        <div
          class="col-md-6 col-12"
        >
          <div
            class="row"
          >
            <div
              class="col-12"
            >
              <div
                class="justify-content-md-center mb-4 nav"
              >
                <div
                  class="mx-1 px-1 nav-item"
                >
                  <a
                    class="nav-link"
                    data-rb-event-key="/login"
                    href="/login"
                  >
                    Sign In
                  </a>
                </div>
                <div
                  class="mx-1 px-1 nav-item"
                >
                  <a
                    class="nav-link"
                    data-rb-event-key="/shipping"
                    href="/shipping"
                  >
                    Shipping
                  </a>
                </div>
                <div
                  class="mx-1 px-1 nav-item"
                >
                  <a
                    class="nav-link"
                    data-rb-event-key="/payment"
                    href="/payment"
                  >
                    Payment
                  </a>
                </div>
                <div
                  class="mx-1 px-1 nav-item"
                >
                  <a
                    aria-disabled="true"
                    class="nav-link disabled"
                    href="#"
                    role="button"
                    tabindex="-1"
                  >
                    Place Order
                  </a>
                </div>
              </div>
            </div>
          </div>
          <h1>
            Payment Method
          </h1>
          <form
            class=""
          >
            <div
              class="form-group"
            >
              <legend
                class="form-label"
              >
                Select Method
              </legend>
              <div
                class="col"
              >
                <div
                  class="form-check"
                >
                  <input
                    checked=""
                    class="form-check-input"
                    id="Paypal"
                    name="paymentMethod"
                    type="radio"
                    value="PayPal"
                  />
                  <label
                    class="form-check-label"
                    for="Paypal"
                    title=""
                  >
                    Paypal or Credit Card
                  </label>
                </div>
                <div
                  class="form-check"
                >
                  <input
                    class="form-check-input"
                    id="Stripe"
                    name="paymentMethod"
                    type="radio"
                    value="Stripe"
                  />
                  <label
                    class="form-check-label"
                    for="Stripe"
                    title=""
                  >
                    Stripe
                  </label>
                </div>
              </div>
            </div>
            <button
              class="my-2 btn btn-primary"
              type="submit"
            >
              Continue
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
